timestampvm/block.go

```diff
@@ -5,52 +5,71 @@ package timestampvm
 
 import (
 	"errors"
+	"fmt"
 	"time"
 
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/choices"
 	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
-	"github.com/ava-labs/avalanchego/vms/components/core"
+	"github.com/ava-labs/avalanchego/utils/hashing"
 )
 
 var (
 	errTimestampTooEarly = errors.New("block's timestamp is earlier than its parent's timestamp")
 	errDatabaseGet       = errors.New("error while retrieving data from database")
-	errDatabaseSave      = errors.New("error while saving block to the database")
 	errTimestampTooLate  = errors.New("block's timestamp is more than 1 hour ahead of local time")
-	errBlockType         = errors.New("unexpected block type")
+	errBlockNil          = errors.New("block is nil")
 
-	_ snowman.Block = &Block{}
+	_ Block = &TimeBlock{}
 )
 
+// Block interface is a wrapper around snowman.Block
+type Block interface {
+	snowman.Block
+	Initialize(bytes []byte, status choices.Status, vm *VM)
+	Data() [dataLen]byte
+}
+
 // Block is a block on the chain.
 // Each block contains:
-// 1) A piece of data (a string)
-// 2) A timestamp
-type Block struct {
-	*core.Block `serialize:"true"`
-	Data        [dataLen]byte `serialize:"true"`
+// 1) ParentID
+// 2) Height
+// 3) Timestamp
+// 4) A piece of data (a string)
+type TimeBlock struct {
+	PrntID ids.ID        `serialize:"true" json:"parentID"`  // parent's ID
+	Hght   uint64        `serialize:"true" json:"height"`    // This block's height. The genesis block is at height 0.
+	Tmstmp int64         `serialize:"true" json:"timestamp"` // Time this block was proposed at
+	Dt     [dataLen]byte `serialize:"true" json:"data"`      // Arbitrary data
+
+	id     ids.ID         // hold this block's ID
+	bytes  []byte         // this block's encoded bytes
+	status choices.Status // block's status
+	vm     *VM            // the underlying VM reference, mostly used for state
 }
 
 // Verify returns nil iff this block is valid.
 // To be valid, it must be that:
 // b.parent.Timestamp < b.Timestamp <= [local time] + 1 hour
-func (b *Block) Verify() error {
-	// Check to see if this block has already been verified by calling Verify on the
-	// embedded *core.Block.
-	// If there is an error while checking, return an error.
-	// If the core.Block says the block is accepted, return accepted.
-	if accepted, err := b.Block.Verify(); err != nil || accepted {
-		return err
+func (b *TimeBlock) Verify() error {
+	if b == nil {
+		return errBlockNil
```

Comment:
I know this check is in other VMs as well - but I don't think it's actually needed.